docs/cloud/execution/fargate_task_environment.md

```diff
@@ -0,0 +1,178 @@
+
+```
+cluster                     string
+taskDefinition              string
+count                       integer
+startedBy                   string
+group                       string
+placementConstraints        list
+placementStrategy           list
+platformVersion             string
+networkConfiguration        dict
+tags                        list
+enableECSManagedTags        boolean
+propagateTags               string
+```
+
+All of these kwargs will be loaded and stored upon initialization of the Environment. It will never be sent to Prefect Cloud and will only exist inside your Flow's Docker storage.
+
+:::tip Task IAM Roles
+Users have seen great performance in using [Task IAM Roles](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-iam-roles.html) for their Flow execution.
+:::
+
+#### Setup
+
+The Fargate Task Environment setup step is responsible for registering the Fargate Task if it does not already exist. First it checks for the existence of a task definition based on the `family` that was provided at initialization of this Environment. If the task definition is not found then it is created. This means that if a Flow is run multiple times the task definition will only need to be created once.
+
+#### Execute
+
+Create a new Fargate Task with the configuration provided at initialization of this Environment. That task is responsible for running the Flow.
+
+#### Task Spec Configuration
+
+There are a few caveats to using the Fargate Task Environment that revolve around the provided boto3 kwargs. In the `containerDefinitions` that you provide the **first container** listed will be the container that is used to run the Flow. This means that the first container will always be overridden during the `setup` step of this Environment.
+
+```python
+containerDefinitions=[
+    {
+        "name": "flow",
+        "image": "image",
+        "command": [],
+        "environment": [],
+        "essential": True,
+    }
+],
+```
+
+In the above dictionary the `flow` container will have a few aspects changed during setup.
+
+- `name` will become _flow-container_
+- `image` will become the _registry_url/image_name:image_tag_ of your Flow's storage
+- `command` will take the form of:
+
+```python
+[
+    "/bin/sh",
+    "-c",
+    "python -c 'from prefect.environments import FargateTaskEnvironment; FargateTaskEnvironment().run_flow()'",
+]
+```
+
+- `environment` will have some extra variables automatically appended to it for Cloud-based Flow runs:
+
+```
+PREFECT__CLOUD__GRAPHQL
+PREFECT__CLOUD__USE_LOCAL_SECRETS
+PREFECT__ENGINE__FLOW_RUNNER__DEFAULT_CLASS
+PREFECT__ENGINE__TASK_RUNNER__DEFAULT_CLASS
+PREFECT__LOGGING__LOG_TO_CLOUD
+```
+
+All other aspects of your `containerDefinitions` will remain untouched. In some cases it is easiest to simply use a dummy first container similar to the code block above.
```

Comment:
Could we highlight some cases where it might be easier or specify why a "dummy" container would be preferable?